projects/storefrontlib/src/cms-components/cart/cart-coupon/cart-coupon-anchor/cart-coupon-anchor.component.spec.ts

```diff
@@ -0,0 +1,47 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { I18nTestingModule } from '@spartacus/core';
+import { CartCouponAnchorComponent } from './cart-coupon-anchor.component';
+import { CartCouponAnchorService } from './cart-coupon-anchor.service';
+
+describe('CartCouponAnchorComponent', () => {
+  let component: CartCouponAnchorComponent;
+  let fixture: ComponentFixture<CartCouponAnchorComponent>;
+
+  let cartCouponAnchorService;
+  let spyEmitter;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [I18nTestingModule],
+      declarations: [CartCouponAnchorComponent],
+      providers: [CartCouponAnchorService],
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(CartCouponAnchorComponent);
+    component = fixture.componentInstance;
+    cartCouponAnchorService = TestBed.get(CartCouponAnchorService);
+    spyEmitter = jasmine.createSpyObj('EventEmitter', ['emit']);
+    spyOn(cartCouponAnchorService, 'getEventEmit').and.returnValue(spyEmitter);
+    spyOn(component, 'sendScrollEvent').and.callThrough();
```

Comment:
remove this line, should not know this method in component.

Reply:
is it better to create a new method is service to call getEventEmit().emit(anchor);?
to decouple the EventEmitter, we don't need know the implementation of the service

Reply:
Won't fix: the elements are in different component without parent relationship, so have to subscribe this event